### PR TITLE
openstack: Add application credentials periodic

### PR DESCRIPTION
**Pull request by pierreprinetti**
Enable a periodic job in OpenStack to test installation with application
credentials.

Implements [OSASINFRA-2720](https://issues.redhat.com/browse/OSASINFRA-2720)

### CODE REVIEW

**Comment by mandre** [comment]
Failed to rollout the prometheus operator?
```
W1101 10:49:55.582873       1 tasks.go:71] task 4 of 15: Updating Prometheus-k8s failed: waiting for Prometheus object changes failed: waiting for Prometheus openshift-monitoring/k8s: expected 2 replicas, got 0 updated replicas
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/23126/rehearse-23126-periodic-ci-openshift-release-master-nightly-4.10-e2e-openstack-appcreds/1455103755543384064/artifacts/e2e-openstack-appcreds/gather-extra/artifacts/pods/openshift-monitoring_cluster-monitoring-operator-5cf79758bd-r6x74_cluster-monitoring-operator.log

/retest

**Comment by pierreprinetti** [comment]
rebased

**Comment by pierreprinetti** [comment]
/hold
adding a debug job

**Comment by pierreprinetti** [comment]
The in-tree cloud provider [does not support application credentials](https://github.com/openshift/kubernetes/blob/fdb6e2b/staging/src/k8s.io/legacy-cloud-providers/openstack/openstack.go#L164-L180) (thanks @mandre).

The job has been changed to enable the external cloud provider.

**Comment by pierreprinetti** [comment]
/hold cancel
Let's wait for the rehearsal job though.

**Comment by pierreprinetti** [comment]
Prometheus again?
/retest

**Comment by mandre** [comment]
> Prometheus again? /retest

From [the must-gather's event filter](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/23126/rehearse-23126-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.10-e2e-openstack-appcreds/1471066161129984000/artifacts/e2e-openstack-appcreds/gather-must-gather/artifacts/event-filter.html):

```
11:55:33 (x145) | openshift-monitoring | persistentvolume-controller | prometheus-data-prometheus-k8s-0 | ExternalProvisioning | waiting  for a volume to be created, either by external provisioner  "cinder.csi.openstack.org" or manually created by system administrator
```

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 1]
```diff
@@ -19981,6 +19981,81 @@ periodics:
...
+  - base_ref: master
+    org: openshift
+    repo: release
+  interval: 24h
+  labels:
+    ci-operator.openshift.io/variant: nightly-4.10
+    ci.openshift.io/generator: prowgen
+    job-release: "4.10"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-release-master-nightly-4.10-e2e-openstack-appcreds
```
[the remark this answers is not in the document]
Comment (pierreprinetti) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml]: Now that the test runs on TP and has a chance to succeed, I have added the notifier :) Thanks

**Comment by mandre** [comment]
/remove-lifecycle stale
We're waiting for a rebase of upstream CAPO in MAPO.

**Comment by pierreprinetti** [comment]
The nightly used in tests is old; the MAPO image in the payload does not contain the required CAPO bump.

This has to wait until a new nightly is built.

**Comment by pierreprinetti** [comment]
The Cinder issues [do not seem to be](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/23126/rehearse-23126-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.11-e2e-openstack-appcreds/1522194162584129536) fixed. This failure requires investigation.

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.12.yaml [hunk 2]
```diff
@@ -136,6 +136,15 @@ tests:
         group\|The OpenStack platform creates Control plane nodes in a server group
       TEST_SUITE: experimental/reliability/minimal
     workflow: openshift-e2e-openstack-upi
+- as: e2e-openstack-appcreds
+  interval: 24h
+  steps:
+    cluster_profile: openstack-vexxhost
+    env:
+      APPLICATION_CREDENTIALS: "true"
+      # these tests are skipped since they use cinderclient which does not support application credentials
```
Comment (pierreprinetti) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.12.yaml:145]: Prow doesn't like comments in config.

Running `make ci-operator-config` (part of `make update`) removes them.

Comment (stephenfin) [reply ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.12.yaml]: Done

**Review by mandre — COMMENTED** [review]
Glad we finally went to the bottom of why the cinder tests were failing.
/lgtm
/approve

**Pull request merged**